### ISSUE TITLE
Unable to override SiteConfig.Title in some parts of the CMS

### ISSUE THREAD

**Issue by alessandromarotta** [silverstripe/silverstripe-framework#8215]
## Affected Version

silverstripe-themes/simple     dev-master e4b3aa2 The SilverStripe simple theme (default SilverStripe 3 theme)
silverstripe/lumberjack        2.0.0              A module to make managing pages in a GridField easy without losing any of the functionality that you're used to i...
silverstripe/recipe-cms        1.1.2              SilverStripe recipe for fully featured page and asset content editing
silverstripe/recipe-plugin     1.3.0              Helper plugin to install SilverStripe recipes
tractorcow/silverstripe-fluent 4.1.2              Simple localisation for Silverstripe

## Description

I need to override the getTitle function of the SiteConfig, so I writed that function in my SiteConfigExtension

namespace alesmaweb\core\Extensions;

use SilverStripe\ORM\DataExtension;

class amwSiteConfigExtension extends DataExtension {

   public function getTitle() {
      return "myString";
   }

}

This function does nothing.

Instead, if I write this function:

 public function Title() {
      return "myString";
 }

It returns "myString" in frontend and in backend only here

![image](https://user-images.githubusercontent.com/17084152/41837092-cf804196-785c-11e8-91fa-e815cfd39825.png)

Here, however, not even

![image](https://user-images.githubusercontent.com/17084152/41837243-38ee39ee-785d-11e8-9020-493276857538.png)

Please, how can I solve? I think that it's a bug, isn´t?

Thank you!

**Comment by kinglozzer** [comment]
There’s a config setting for this https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L1779:

```yml
SilverStripe\Admin\LeftAndMain:
  application_name: 'MyString'
```

**Comment by alessandromarotta** [comment]
Hi! thanks for the reply but I think that doesn't solve the problem. I need to get the value from the SiteConfig table (field "Title") and not from the configuration file.

**Comment by kinglozzer** [comment]
In that case, you’ll need to update the config value in PHP - after the database is ready - with something like:

```php
use SilverStripe\Admin\LeftAndMain;
use SilverStripe\SiteConfig\SiteConfig;

// ...

$siteConfig = SiteConfig::current_site_config();
Config::modify()->set(LeftAndMain::class, 'application_name', $siteConfig->Title);
```

I’d probably put that in an extension applied to `LeftAndMain`, using the `init` hook here: https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L733.

If you’re still stuck, I’d suggest asking on the [forums](http://www.silverstripe.org/community/forums/) or [Slack channel](https://silverstripe-users.signup.team/) :)

**Comment by kinglozzer** [comment]
Oh, I see. It looks like you’re trying to translate the title? If that’s the case, it’s probably better to apply `FluentExtension` to `SiteConfig` and add translations that way rather than storing them as JSON.

As `Title` is a database field in `SiteConfig`, creating a method named `Title()` usually would work (but is not advised). The title at the top of the list of pages (your second screenshot) is hard-coded to fetch the database value, not the template-rendered value here: https://github.com/silverstripe/silverstripe-cms/blob/4/code/Controllers/CMSMain.php#L2225-L2235. If that’s the approach you choose to take, you’ll need to hook into that `updateCMSTreeTitle` extension point.

**Comment by alessandromarotta** [comment]
Yes, exacty, I want to translate Title and Tagline fields. For Tagline all is good but for the Title I found out that problem. I just know the I'm able to translate the title (and other DataObject) with Fluent but in this case I would not create a new table for only two fields. That's all.

P.S: I tried with the updateCMSTreeTitle but nothing happend. I see that the title in the SiteTree page is generated by an ajax function (vendor\silverstripe\cms\templates\SilverStripe\CMS\Controllers\Includes\CMSMain_PageList.ss)
